src/rest-core-types.h

```diff
@@ -93,5 +93,10 @@ void rest_notif_deregistration_delete(rest_notif_deregistration_t *deregistratio
 
 int rest_notif_deregistration_set(rest_notif_deregistration_t *deregistration, const char *name);
 
+
```

Comment:
My suggestion is to have the following or similar API:
```C
/*
 * Decodes base64 string into binary buffer and calculates it's length.
 * base64_string [in] - a null-terminated base64 string.
 * data [out] - pointer to a buffer, can be NULL.
 * length [in/out] - the length of the data buffer (in) / the length of binary base64 data (out).
 * Returns 0 on success, negative value on error:
 *       BASE64_ERR_MEM (-2?) when data is NULL or data length is less than the binary base64 data.
 */
int base64_decode(const char *base64_string, uint8_t *data, size_t *length);
```

Maybe encode is also worth updating?
```C
/*
 * Encodes binary data into base64 string and calculate it's length.
 * data [in] - binary data to be encoded
 * length [in] - length of the binary data
 * base64_string [out] - pointer to a string buffer, can be NULL
 * base64_length [in/out] - pointer to variable storing length of the string buffer (in) / the length of encoded base64 string.
 * Returns 0 on success, negative value on failure.
 */
int base64_encode(const uint8_t *data, size_t length, char *base64_string, size_t *base64_length);
```

What do you think?